hbase-coprocessor/src/main/java/org/apache/omid/transaction/CellSkipFilterBase.java

```diff
@@ -53,21 +53,21 @@ public CellSkipFilterBase(Filter filter) {
      */
     private boolean skipCellVersion(Cell cell) {
         return skipColumn != null
-        && CellUtil.matchingRow(cell, skipColumn.getRowArray(), skipColumn.getRowOffset(),
+        && PrivateCellUtil.matchingRows(cell, skipColumn.getRowArray(), skipColumn.getRowOffset(),
```

Comment:
Don't use PrivateCellUtil anywhere, it's InterfaceAudience.Private.
Use the methods in the Public CellUtil.

Reply:
I agree that InterfaceAudience.Private methods should be avoided. Nonetheless, I used it here because the class already has a KeyValue field which is also InterfaceAudience.Private. 
An alternative solution could be to use Cell instead of KeyValue, and maybe CellUtil.matchingRows(Cell, Cell) instead of this call. How do you think? How should line 73 be changed in this case?

Reply:
Yes.
If you need to copy some code from the IA.Private classes, please add a comment to the copied code that explains why it was copied, and where is the original code.

Reply:
@Aarchy - in general we've been trying to replace uses of KeyValue in Phoenix and its related projects with Cell and its related public interfaces like CellUtil where possible.

Reply:
With @stoty we came up with the solution to duplicate the FirstOnRowCell class (and createFirstOnRow function) from PrivateCellUtil to the Omid codebase. However, this class has several IA.Private base classes. Could it be derived directly from Cell or are the base classes also necessary in Omid? Or is there anything we missed in CellUtil to replace the KeyValueUtil.createFirstOnRow call?

Reply:
Most of the server-side HBase Cell API is Private, which I think is a mistake.
It should be Limited for coprocessor usage.

There is no way Phoenix can operate without those classes.

I think that in this case, making an exception for PrivateCellUtil is fine.

WDYT @gjacoby126  ?

Reply:
@stoty @Aarchy - Agreed. I also don't see an alternative to using an IA.Private class here. Given the choice between KeyValueUtil / KeyValue, and the PrivateCellUtil's createFirstOnRow function, I think the latter's the better option. 

It keeps the interface encapsulation around Cell and doesn't require Phoenix code to make assumptions about whether the Cell is an array-backed on-heap object or an off-heap byte-buffer backed object. 

Rushabh Shah and I once asked to allow PrivateCellUtil to be IA.LimitedPrivate(COPROC) on HBASE-25328, which the HBase community was not willing to do.  Instead they were willing to expose new LimitedPrivate helper methods in HBase that called particular IA.Private methods we needed. Probably not a good time with HBase 2.5 about to release, but that's something we could pursue in the future. 

This does of course mean we're accepting the risk that this method can change arbitrarily with every HBase upgrade.